oml/functional/knn.py

```diff
@@ -0,0 +1,78 @@
+from typing import List, Optional, Tuple
+
+import torch
+from torch import FloatTensor, LongTensor
+
+from oml.const import BS_KNN
+from oml.utils.misc_torch import pairwise_dist
+
+
+def batched_knn(
```

Comment:
I think this func has nothing with KNN. It's _exact_ calculation of distance matrix with truncating. However, I don't know how to call it...

Reply:
it's the same as kNN from sklearn:
```
    from sklearn.neighbors import NearestNeighbors
    knn = NearestNeighbors(algorithm="auto", p=2)
    knn.fit(features_galleries)
    dists, ii_closest = knn.kneighbors(features_queries, n_neighbors=top_k, return_distance=True)
```

By the way, did not you mean aNN (approximate NN)? But it's not what we are doing here